policy/file/source_test.go

```diff
@@ -51,3 +55,158 @@ func TestSource_getFilePolicyID(t *testing.T) {
 		})
 	}
 }
+
+func testFileSource(t *testing.T, dir string) (*Source, []policy.PolicyID) {
+	t.Helper()
+	src := NewFileSource(
+		hclog.Default(),
+		dir, // should contain real policy files.
+		policy.NewProcessor(
+			&policy.ConfigDefaults{
+				DefaultEvaluationInterval: time.Second,
+				DefaultCooldown:           time.Second},
+			[]string{},
+		),
+	)
+	s := src.(*Source)
+	ids, err := s.handleDir() // populate idMap and policyMap
+	if err != nil {
+		t.Fatalf("error from handleDir: %v", err)
+	}
+	if len(ids) == 0 {
+		t.Fatalf("uh oh, no policies in %v", dir)
+	}
+	return s, ids
+}
+
+func TestSource_MonitorPolicy(t *testing.T) {
+	s, ids := testFileSource(t, "./test-fixtures")
+	pid := ids[0]
+
+	// running MonitorPolicy() twice should have the same result.
+	for _, n := range []string{"round one", "round two"} {
+
+		// not using sub-tests here, because if round one fails,
+		// there's no sense running round two.
+		// this little helper gives context for any failures.
+		fatal := func(msg string, args ...any) {
+			t.Helper()
+			t.Fatalf(n+": "+msg, args...)
```

Comment:
Probably not too relevant for a test, but I think this can cause the `go s.MonitorPolicy(ctx, req)` goroutine to leak. You could register a `t.Cleanup(cancel)` below to make sure the context is cancelled when the test exits.